### DISCUSSION TITLE
Guest integration and VM additions support

### DISCUSSION THREAD

**Discussion by atauenis** [86Box/86Box#1727] [Ideas]
I think, it will be great if 86Box will have support for guest OS integration. I've used for may years MS Virtual PC 2007 due to it its Virtual Machine Additions for DOS/Win9x/XP/OS2. It allows file copy between guest and host OS via drag&drop, mouse pointer integration, shared folders support, clipboard sharing, and guest OS soft shutdown.

Now that VM does not works in modern systems, so I've moved to 86Box and is feeling absence of VM Additions support. They was very useful for fast copying files/data between virtual and host system.

It is not need to make support with exactly MS/Connectix Additions, but their features are a good thing.

**Comment by ghost** [comment]
On a discord server, it might be not possible for reasons. Might aswell just wire up a Pcap and network between these two according to explorerexe.

**Comment by atauenis** [comment]
Enabling SMBv1 on modern Windows makes system insecure, so is not a good idea. Using FTP and similar protocols does not allow running EXE files from network, so is bit difficult. Currently I'm using WinImage and direct copy to/from virtual disk images, which requires VM shutdown on every time. But all copied software works natively after this procedure, just like running from shared folder.

**Comment by dhrdlicka** [comment]
A full blown set of additions similar to other virtualization software is not really feasible, as it would require us to create new drivers or reverse existing ones. We don't really have resources for either.

The closest thing we could do without any need for guest drivers would be something similar to VVFAT or VISO, where you'd select some files and the emulator would automatically generate the image without the need for external software.

**Comment by atauenis** [comment]
I have understand you. Theoretically, this can be do using only user-mode software via COM port. At least clipboard synchronization, mouse pointer moving and some file transfer seems to be easy implementable over it.

In some future I will try to write such app (huh, in Visual Basic 3-6 :) ) and publish its interface. Probably it would be possible later to port it to C and then to 86Box.

Shared folder support is a some harder thing. Virtual PC implements it as a virtual device with own driver, which mounts as a virtual drive only after driver load. VirtualBox uses similar way, but works over pseudo-Samba (\\workgroup\vboxsvr virtual server). An something like VVFAT from QEMU seems an interesting idea, as this will allow even OS installing through a boot floppy and such virtual "drivectory" visible as IDE disk. But this way limits amount of mounted directories to amount of free IDE/ESDI/MFM ports (e.g. 4 maximum), so I'm don't like this method.

**Comment by OBattler** [comment]
There is an alternative for file transfers - emulate a ZIP drive (look in Settings). Then you can just mount and unmount, with no need to shut down the VM. A Magneto-Optical drive might work for this purpose as well (again, look in Settings).
